### PR TITLE
Kushki: Add new gateway

### PR DESCRIPTION
**Pull request by shasum**
@davidsantoso

### CODE REVIEW

**Comment by davidsantoso** [comment]
@duff @rwdaigle @jasonwebster would appreciate if we could get your eyes on this one. The Kushki gateway requires a `DELETE` request with a request body to run their `void` so we're adding an additional request type. Wondering if you have any thoughts on naming, general implementation, other thoughts around this kind of addition?

File: lib/active_merchant/delete_with_body.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+require 'net/http'
+require 'net/https'
+
+module ActiveMerchant
```
Comment (rwdaigle) [inline lib/active_merchant/delete_with_body.rb]: Not gonna act like I'd build a library that uses class constants to define functionality, but at least it's convenient 👍

**Comment by davidsantoso** [comment]
@duff @jasonwebster another quick ping to see if either of you have any thoughts. No worries if not, just wanna make sure things gets a couple more eyes to be sure we're 👍 considering it's got a new HTTP `DELETE` request that allows a body.

**Comment by duff** [comment]
👍 👍

File: lib/active_merchant/billing/gateways/kushki.rb [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+        post = {}
+        add_reference(post, authorization, options)
+        add_invoice(action, post, amount, options)
+
+        commit(action, post)
+      end
+
+      def add_invoice(action, post, money, options)
+        if action == "tokenize"
+          post[:totalAmount] = amount(money).to_f
```
Comment (jasonwebster) [inline lib/active_merchant/billing/gateways/kushki.rb]: Should this include a precision?

File: lib/active_merchant/connection.rb [hunk 3]
```diff
@@ -77,6 +77,9 @@ def request(method, body, headers = {})
               # very unambiguously does not.
               raise ArgumentError, "DELETE requests do not support a request body" if body
               http.delete(endpoint.request_uri, headers)
+            when :delete_with_body
+              debug body
```
Comment (jasonwebster) [inline lib/active_merchant/connection.rb]: Oh, nevermind. They all do this. Sorry

File: lib/active_merchant/billing/gateways/kushki.rb (hunk 2 above)
Comment (shasum) [inline lib/active_merchant/billing/gateways/kushki.rb]: @jasonwebster The `amount()` method does that internally.

https://github.com/activemerchant/active_merchant/blob/master/lib/active_merchant/billing/gateway.rb#L254

**Comment by davidsantoso** [comment]
Alrighty, looks like no major objections. I'm going to merge this in.

**Comment by davidsantoso** [comment]
@shasum thanks!

**Pull request closed without merging**